### DISCUSSION TITLE
Upload audio/video to browser and generate subtitles that display as media is played

### DISCUSSION THREAD

**Discussion by goudbor** [openai/whisper#1323] [Show and tell]
Wrote a little app that can be run locally to generate subtitles for an audio/video file and display them as you play the media in the browser.  Hope you like!

[Frontend](https://github.com/jojeyh/subtitler)
[Backend](https://github.com/jojeyh/subtitler-backend)

**Comment by dgoryeo** [comment]
@jojeyh , do I understand right that this is a real-time sub generator? Would be nice to plug it into VLC or Potplayer.

**Reply by goudbor** [reply]
@dgoryeo This is not real-time unfortunately!  On uploading the file it generates all subtitles for entire file and then lines them up with play.  I'm working on real-time solutions for a use case I have, but not sure why you would want it for VLC?  You could have a plugin for VLC that generates subtitles and adds them to VLC based on timestamps.

**Reply by goudbor** [reply]
@dgoryeo Do you have experience with VLC plugins? I'd be willing to work on a little project that will generate english subtitles in VLC?

**Reply by dgoryeo** [reply]
Hi @jojeyh , just saw this --sorry for late replay. I'm afraid I do not have experience with VLC plugins.
